### ISSUE TITLE
[Question] How to define `$ref` to definitions in other files when overwriting definition

### ISSUE THREAD

**Issue by archerzz** [Azure/autorest#4237]
Before asking the question:

- [x] have you checked the [faq](https://github.com/Azure/autorest/tree/main/docs/faq.md), the documentation in the [docs folder](https://github.com/Azure/autorest/tree/main/docs) and couldn't find the information there
- [x] have you checked existing issues for a similar question?

## Problem

I have a [definition](https://github.com/Azure/azure-rest-api-specs/blob/83c0c488dfe3f49d7c666511d5d6714ec6b4502f/specification/network/resource-manager/Microsoft.Network/stable/2021-02-01/ddosProtectionPlan.json#L353) which has excessive properties. So I want to re-write it as follow, using `where-model` directives.
```
 - where-model: DdosProtectionPlan
    transform: >
      return {
        "description": "A DDoS protection plan in a resource group.",
        "x-ms-azure-resource": true,
        "properties": {
          "properties": {
            "x-ms-client-flatten": true,
            "$ref": "#/definitions/DdosProtectionPlanPropertiesFormat",
            "description": "Properties of the DDoS protection plan."
          },
          "etag": {
            "readOnly": true,
            "type": "string",
            "description": "A unique read-only string that changes whenever the resource is updated."
          }
        },
        "allOf": [
            {
              "$ref": "./network.json#/definitions/Resource" ## PROBLEM!
            }
        ]
      }
```

The problem is that when using autorest to generate the codes, I got the following error. [Note]: If I modify the original swagger JSON file, the generation is OK.

>  Error: $ref to original location './network.json#/components/schemas/Resource' is not found in the new refs collection

I tried many alternatives, and it seems that when `where-model` directive is being processed, the reference key is changed to a global absolute value. If I changed the `$ref` to `file:///C:/Users/mingzhehuang/workspaces/azure/azure-rest-api-specs/specification/network/resource-manager/Microsoft.Network/stable/2021-02-01/network.json#/definitions/Resource`, the generation is fine.

## Question

Is there a way I can safely refer to definitions in other files when overwriting the definition? Thanks.

**Comment by timotheeguerin** [comment]
Yeah, I think this is due to how references are expanded at the beginning but not again later. 
you can use `$documentPath` to make the full reference

"$ref": "$documentPath/../network.json#/definitions/Resource"

We could maybe do another crawl of the reference after the transform step

**Comment by archerzz** [comment]
Thanks @timotheeguerin !

It would be better to re-calculate the `$ref` if the given value is not found. That will make the customization much easier.

**Comment by archerzz** [comment]
@timotheeguerin Do you want me close this issue, or keep it open as a feature request?
